src/utils/viewer/viewer.cpp

```diff
@@ -481,8 +521,16 @@ void Viewer::drawEvent() {
   }
 
   // TODO: enable other sensors to be displayed
-  simulator_->displayObservation(defaultAgentId_, "rgba_camera");
-  uint32_t visibles = renderCamera_->getPreviousNumVisibileDrawables();
+  uint32_t visibles = 0;
+  if (objectPickingHelper_->isObjectPicked()) {
+    visibles = displayObservatationWithPickedObject();
+  } else {
+    simulator_->displayObservation(defaultAgentId_, "rgba_camera");
+    visibles = renderCamera_->getPreviousNumVisibileDrawables();
+  }
+  // Immediately bind the main buffer back so that the "imgui" below can work
+  // properly
+  Mn::GL::defaultFramebuffer.bind();
```

Comment:
After the changes I proposed above, the contents of `displayObservationWithPickedObject()` could be pasted directly here, deduplicating the code quite a lot:

```suggestion
  // draw, but don't blit it to the default framebuffer just yet
  simulator_->drawObservationTo(defaultAgentId_, "rgba_camera", nullptr);
  visibles = renderCamera_->getPreviousNumVisibileDrawables();

  // the render target should be always here
  esp::gfx::RenderTarget* sensorRenderTarget =
      simulator_->getRenderTarget(defaultAgentId_, "rgba_camera");
  CORRADE_INTERNAL_ASSERT(sensorRenderTarget);
  
  // draw the picked object to the same frame buffer for proper depth test
  if (objectPickingHelper_->isObjectPicked()) {
    sensorRenderTarget->renderReEnter();    
    
    Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::Blending);

    // blend the picked object on top of the existing contents
    esp::gfx::RenderCamera::Flags flags;
    if (simulator_->isFrustumCullingEnabled()) {
      flags |= esp::gfx::RenderCamera::Flag::FrustumCulling;
    }
    renderCamera_->draw(objectPickingHelper_->getDrawables(), flags);

    Mn::GL::Renderer::disable(Mn::GL::Renderer::Feature::Blending);
    
    // bind the main buffer back so the rest works properly
    sensorRenderTarget->renderExit();    
    Mn::GL::defaultFramebuffer.bind();
  }
  
  // finally, blit the observation (and optional picked object on top) to the 
  // default framebuffer
  sensorRenderTarget->blitRgbaTo(Mn::GL::defaultFramebuffer);
```